src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarTownEventListener.java

```diff
@@ -434,11 +438,17 @@ public void onTownStatusScreen(TownStatusScreenEvent event) {
 	            }
 	        } else {
 	            if(!SiegeController.hasActiveSiege(town)
-	            	&& System.currentTimeMillis() < TownMetaDataController.getSiegeImmunityEndTime(town)) {
+	            	&& (System.currentTimeMillis() < TownMetaDataController.getSiegeImmunityEndTime(town)
+					|| TownMetaDataController.getSiegeImmunityEndTime(town) == -1l)) {
 	                //Siege:
 	                // > Immunity Timer: 40.8 hours
 	                out.add(Translation.of("status_town_siege"));
-	                String time = TimeMgmt.getFormattedTimeValue(TownMetaDataController.getSiegeImmunityEndTime(town)- System.currentTimeMillis()); 
+					String time;
+	                if (TownMetaDataController.getSiegeImmunityEndTime(town) != -1l) {
+						time = TimeMgmt.getFormattedTimeValue(TownMetaDataController.getSiegeImmunityEndTime(town)- System.currentTimeMillis());
+					} else {
+	                	time = "Permanent";

```

Comment:
:construction: 
- You need to get the translation for `msg_permanent` here, rather than hardcode the word "permanent"

Reply:
:construction: 
This still needs to be done.